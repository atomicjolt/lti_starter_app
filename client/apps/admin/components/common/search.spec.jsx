import React from 'react';
import TestRenderer from 'react-test-renderer';
import Search from './search';

describe('common search', () => {
  let result;
  let search;
  let props;

  beforeEach(() => {
    search = false;
    props = {
      search: () => { search = true; },
    };
    result = TestRenderer.create(<Search {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('has search input', () => {
    const input = instance.findByType('input');
    expect(input).toBeDefined();
    expect(input.props().type).toBe('text');
    expect(input.props().placeholder).toBe('Search...');
  });

  it('search input changes', () => {
    expect(search).toBeFalsy();
    instance.findByType('input').simulate('change', { target: { value: 'new value' } });
    expect(search).toBeTruthy();
  });
});
