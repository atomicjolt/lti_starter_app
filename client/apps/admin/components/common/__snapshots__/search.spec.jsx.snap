// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common search matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <search
    search={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="Search..."
          type="text"
        />,
        <i
          className="i-search"
        />,
      ],
      "className": "c-search c-search--small",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Search...",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "i-search",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="Search..."
            type="text"
          />,
          <i
            className="i-search"
          />,
        ],
        "className": "c-search c-search--small",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Search...",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "i-search",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
